web-app/app/tricorder/Configure.tsx

```diff
@@ -1,52 +1,53 @@
-import { useEffect } from "react";
-import { buttonClass } from "./classes";
+export type Config = {
+  antimatter?: boolean;
+  slingshot?: boolean;
+  stembolts?: boolean;
+};
 
-export function Configure(props: any) {
-  const onChange = props.onChange;
-  const configuration: Record<any, any> = {};
-  useEffect(() => {
-    localStorage.setItem(
-      "__tricorder__configuration",
-      JSON.stringify(configuration)
-    );
-  }, [configuration]);
-  useEffect(() => {
-    const newConfiguration = JSON.parse(
-      localStorage.getItem("__tricorder_configuration") ?? "{}"
-    );
-    Object.keys(newConfiguration).map((configKey) => {
-      configuration[configKey] = newConfiguration[configKey];
-    });
-  }, []);
+export function Configure({
+  config,
+  onChange,
+}: {
+  config: Config;
+  onChange: (configKey: keyof Config, value: boolean) => void;
+}) {
   return (
     <div>
       <button
-        className={buttonClass}
+        className={`lcars-button ${config["slingshot"] ? "on" : ""}`}
         onClick={() => {
-          configuration.slingshot = true;
-          onChange(configuration);
+          onChange("slingshot", !config.slingshot);
```

Comment:
Rather than manipulating the configuration object, we're instead letting the implementing component decide how configuration state is handled. We inform them only of the specific changes.